traceon/field.py

```diff
@@ -707,5 +896,130 @@ def get_low_level_trace_function(self):
+        elec, mag = self.is_electrostatic(), self.is_magnetostatic()
+        
+        if elec and not mag:
+            return self.electrostatic_field_at_point(point)
+        elif not elec and mag:
+            return self.magnetostatic_field_at_point(point)
+        
+        raise RuntimeError("Cannot use field_at_point when both electric and magnetic fields are present, " \
+            "use electrostatic_field_at_point or magnetostatic_field_at_point")
+
+    def potential_at_point(self, point):
+        elec, mag = self.is_electrostatic(), self.is_magnetostatic()
+
+        if elec and not mag:
+            return self.electrostatic_potential_at_point(point)
+        elif not elec and mag:
+            return self.magnetostatic_potential_at_point(point)
+        
+        raise RuntimeError("Cannot use potential_at_point when both electric and magnetic fields are present, " \
+            "use electrostatic_potential_at_point or magnetostatic_potential_at_point")
+        
+    def electrostatic_field_at_point(self, point):
+        return np.sum([f.electrostatic_field_at_point(point) for f in self.fields], axis=0)
+
+    def magnetostatic_field_at_point(self, point):
+        return np.sum([f.magnetostatic_field_at_point(point) for f in self.fields], axis=0)
+    
+    def current_field_at_point(self, point):
+        return np.sum([f.current_field_at_point(point) for f in self.fields], axis=0)
+
+    def electrostatic_potential_at_point(self, point):
+        return sum([f.electrostatic_potential_at_point(point) for f in self.fields])
+
+    def magnetostatic_potential_at_point(self, point):
+        return sum([f.magnetostatic_potential_at_point(point) for f in self.fields])
+    
+    def electrostatic_field_at_local_point(self, point):
+        return self.electrostatic_field_at_point(point)
+
+    def magnetostatic_field_at_local_point(self, point):
+        return self.magnetostatic_field_at_point(point)
+    
+    def current_field_at_local_point(self, point):
+        return self.current_field_at_point(point)
+
+    def electrostatic_potential_at_local_point(self, point):
+        return self.electrostatic_potential_at_point(point)
+
+    def magnetostatic_potential_at_local_point(self, point):
+        return self.magnetostatic_potential_at_point(point)
+    
+    def is_electrostatic(self):
+        return any(f.is_electrostatic() for f in self.fields)
+    
+    def is_magnetostatic(self):
+        return any(f.is_magnetostatic() for f in self.fields)
+
+    def get_tracer(self, bounds):
+        return T.Tracer(self, bounds)
+
+    def __add__(self, other):
+        if isinstance(other, Field):
+            other_fields = other.fields if isinstance(other, FieldSuperposition) else [other]
+            fields_copy = self.fields.copy()
+            for of in other_fields:
+                for i, f in enumerate(fields_copy):
+                    if f._matches_geometry(of):
+                        fields_copy[i] = f + of
```

Comment:
In principle, modifying a list while iterating over it is a bad idea. Consider changing `enumerate(fields_copy)` to `enumerate(self.fields)`